lib/Logger.js

```diff
@@ -49,6 +54,40 @@ class Logger {
         Config.update({ dump: levelName });
     }
 
+    getLogLevel() {
+        return Config.level;
+    }
+
+    getDumpLevel() {
+        return Config.dump;
+    }
+
+    /**
+     *  This function creates a new Logger, from another one,
+     *  changing its submodule field.
+     *
+     *  @param {Logger} baseLogger - Base Logger to create a new one.
+     *  @param {String} submodule - The name of the submodule.
+     *  @return {Logger} - The new logger with the new submodule.
+     */
+    static updateSubmodule(baseLogger, submodule) {
+        return new Logger(baseLogger.name, {
+            level: baseLogger.getLogLevel(),
+            dump: baseLogger.getDumpLevel(),
+        }, submodule);
+    }
+
+    /**
+     *  This factory creates a function which update the submodule of a
+     *  RequestLogger according to a ModuleLogger submodule.
+     *
+     *  @param {Logger} log - Base Logger to create a new one.
+     *  @return {RequestLogger} - The new logger with the new submodule.
+     */
+    static updateRequestLoggerSubmoduleFactory(log) {
```

Comment:
Nope, I'm against the Java-isation with meaningless factories and very long names
This should be a method of a requestLogger instance instead, as we discussed.